src/mumble/pluginAPI/PluginComponents.h

```diff
@@ -0,0 +1,134 @@
+/**
+ * This header file contains definitions of types and other components used in Mumble's plugin system
+ */
+
+#ifndef MUMBLE_PLUGINCOMPONENT_H_
+#define MUMBLE_PLUGINCOMPONENT_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define STATUS_OK 0
+
+enum TalkingState {
+	INVALID=-1,
+	PASSIVE=0,
+	TALKING,
+	WHISPERING,
+	SHOUTING
+};
+
+struct Version {
+	int32_t major;
+	int32_t minor;
+	int32_t patch;
+};
+
+
+typedef enum TalkingState TalkingState_t;
+typedef struct Version Version_t;
+typedef int32_t MumbleConnection_t;
+typedef uint32_t MumbleUserID_t;
+typedef int32_t MumbleChannelID_t;
+typedef int32_t error_t;
```

Comment:
Might make that a enum later too?

Reply:
yes, making an enum here makes more sense than defining each error type using a define. Specially for debugging purposes.

Reply:
Yeah I guess we can do that. My mind has not yet wrapped around the fact that in C we can easily add new values to an enum without breaking existing code (as is the case in e.g. Java).